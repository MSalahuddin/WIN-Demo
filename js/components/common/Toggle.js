import React from 'react';
import { TouchableWithoutFeedback } from 'react-native';
import styled from 'styled-components/native';
import PropTypes from 'prop-types';

import Text, { SIZE, FONT_FAMILY } from './Text';
import { scale, scaleHeight } from '../../platformUtils';
import { color } from '../../styles';

const Container = styled.View`
  flex: 1;
  justify-content: center;
  align-items: center;
`;

const radius = 35;
const backgroundRadius = radius + 4;

const ToggleBorder = styled.View`
  align-items: center;
  justify-content: center;
  background-color: ${color.green};
  border-radius: ${scale(backgroundRadius)};
  width: ${scale(57)};
  height: ${scale(38)};
`;

const ToggleBackground = styled.View`
  align-items: center;
  background-color: ${({ enabled }) => (enabled ? color.profileNavigationSecondary3 : color.silver)};
  border-radius: ${scale(backgroundRadius)};
  flex-direction: row;
  justify-content: ${({ enabled }) => (enabled ? 'flex-end' : 'flex-start')};
  margin-horizontal: ${scale(10)};
  padding-horizontal: ${scale(1)};
  padding-vertical: ${scale(1)};
  width: ${scale(51)};
  height: ${scale(32)};
`;

const ToggleSwitch = styled.View`
  width: ${scale(28)};
  height: ${scale(28)};
  background-color: ${color.white};
  border-width: 0;
  border-radius: ${scale(14)};
`;

const TextWrapper = styled(Text)`
  margin-top: ${scaleHeight(8)};
`;

const Toggle = ({ enabled, onToggle, label }) => {
  return (
    <Container>
      <TouchableWithoutFeedback onPress={onToggle}>
        <ToggleBorder>
          <ToggleBackground enabled={enabled}>
            <ToggleSwitch />
          </ToggleBackground>
        </ToggleBorder>
      </TouchableWithoutFeedback>
      <TextWrapper size={SIZE.XXSMALL} fontFamily={FONT_FAMILY.SEMI_BOLD}>
        {label}
      </TextWrapper>
    </Container>
  );
};

Toggle.defaultProps = {
  label: ''
};

Toggle.propTypes = {
  onToggle: PropTypes.func.isRequired,
  enabled: PropTypes.bool.isRequired,
  label: PropTypes.string
};

export default Toggle;
