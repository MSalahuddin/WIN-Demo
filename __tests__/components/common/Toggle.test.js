import React from 'react';
import { render } from 'react-native-testing-library';

import Toggle from '../../../js/components/common/Toggle';

describe('Component Toggle', () => {
  const onToggle = jest.fn();

  test('renders correctly when it is enabled', () => {
    const output = render(<Toggle enabled onToggle={onToggle} label="a" />);
    expect(output).toMatchSnapshot();
  });

  test('renders correctly when it is enabled=false', () => {
    const output = render(<Toggle enabled={false} onToggle={onToggle} label="a" />);
    expect(output).toMatchSnapshot();
  });
});
