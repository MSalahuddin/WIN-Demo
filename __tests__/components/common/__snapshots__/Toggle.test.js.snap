// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Toggle renders correctly when it is enabled 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#E5E5E5",
          "borderRadius": 58.5,
          "height": 57,
          "justifyContent": "center",
          "width": 85.5,
        },
      ]
    }
  >
    <View
      enabled={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#72D59E",
            "borderRadius": 58.5,
            "flexDirection": "row",
            "height": 48,
            "justifyContent": "flex-end",
            "marginHorizontal": 15,
            "paddingHorizontal": 1.5,
            "paddingVertical": 1.5,
            "width": 76.5,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 21,
              "borderWidth": 0,
              "height": 42,
              "width": 42,
            },
          ]
        }
      />
    </View>
  </View>
  <Text
    alignCenter={false}
    allowFontScaling={true}
    color="#000000"
    fontFamily="SEMI BOLD"
    isUnderlined={false}
    isUppercase={false}
    maxFontSizeMultiplier={1.3}
    size="XXSMALL"
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "Calibre-Semibold",
          "fontSize": 21,
          "justifyContent": "center",
          "textAlign": "left",
          "textDecorationColor": "#000000",
          "textDecorationLine": "none",
          "textDecorationStyle": "solid",
          "textTransform": "none",
        },
        Object {
          "marginTop": 16,
        },
      ]
    }
  >
    a
  </Text>
</View>
`;

exports[`Component Toggle renders correctly when it is enabled=false 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#E5E5E5",
          "borderRadius": 58.5,
          "height": 57,
          "justifyContent": "center",
          "width": 85.5,
        },
      ]
    }
  >
    <View
      enabled={false}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#8B8B8B",
            "borderRadius": 58.5,
            "flexDirection": "row",
            "height": 48,
            "justifyContent": "flex-start",
            "marginHorizontal": 15,
            "paddingHorizontal": 1.5,
            "paddingVertical": 1.5,
            "width": 76.5,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 21,
              "borderWidth": 0,
              "height": 42,
              "width": 42,
            },
          ]
        }
      />
    </View>
  </View>
  <Text
    alignCenter={false}
    allowFontScaling={true}
    color="#000000"
    fontFamily="SEMI BOLD"
    isUnderlined={false}
    isUppercase={false}
    maxFontSizeMultiplier={1.3}
    size="XXSMALL"
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "Calibre-Semibold",
          "fontSize": 21,
          "justifyContent": "center",
          "textAlign": "left",
          "textDecorationColor": "#000000",
          "textDecorationLine": "none",
          "textDecorationStyle": "solid",
          "textTransform": "none",
        },
        Object {
          "marginTop": 16,
        },
      ]
    }
  >
    a
  </Text>
</View>
`;
